scripts/generate.sh

```diff
@@ -14,11 +14,19 @@ done
 
 # jar was downloaded from here https://repo1.maven.org/maven2/io/swagger/codegen/v3/swagger-codegen-cli/3.0.34/
 
-if [[ $VERSION == *"develop"* ]]; then
-  SANITIZED_VERSION=$(sed 's/-develop//g' <<< $VERSION)
-  SANITIZED_VERSION=$(sed 's/\.[0-9]*$//g' <<< $SANITIZED_VERSION)
-  BUILD_VERSION=$(grep -o '[0-9]*$' <<< $VERSION)
-  VERSION="dev-$SANITIZED_VERSION-$BUILD_VERSION"
+VERSION=${VERSION//develop/dev}
+
+if [[ $VERSION =~ ^dev[.-]([0-9]+)[.-]([0-9]+)[.-]([0-9]+)[.-]([0-9]+)$ ]]; then
```

Comment:
Please add comments with the examples for each case so as not to spend time on this next time when we need to edit.